libpod/inspect_data.go

```diff
@@ -0,0 +1,204 @@
+	"github.com/projectatomic/libpod/libpod/driver"
+)
+
+// ContainerData handles the data used when inspecting a container
+type ContainerData struct {
+	Id              string           `json:"Id"`
+	Created         time.Time        `json:"Created"`
+	Path            string           `json:"Path"`
+	Args            []string         `json:"Args"`
+	State           *CtrState        `json:"State"`
+	ImageID         string           `json:"Image"`
+	ImageName       string           `json:"ImageName"`
+	ResolvConfPath  string           `json:"ResolvConfPath"`
+	HostnamePath    string           `json:"HostnamePath"` //not implemented
+	HostsPath       string           `json:"HostsPath"`    //not implemented
+	StaticDir       string           `json:"StaticDir"`
+	LogPath         string           `json:"LogPath"`
+	Name            string           `json:"Name"`
+	RestartCount    int32            `json:"RestartCount"` //not implemented
+	Driver          string           `json:"Driver"`
+	MountLabel      string           `json:"MountLabel"`
+	ProcessLabel    string           `json:"ProcessLabel"`
+	AppArmorProfile string           `json:"AppArmorProfile"`
+	ExecIDs         string           `json:"ExecIDs"` // don't know type yet, not implemented
+	HostConfig      *HostConfig      `json:"HostConfig"`
+	GraphDriver     *driver.Data     `json:"GraphDriver"`
+	SizeRw          int64            `json:"SizeRw,omitempty"`
+	SizeRootFs      int64            `json:"SizeRootFs,omitempty"`
+	Mounts          []specs.Mount    `json:"Mounts"`
+	Config          *CtrConfig       `json:"Config"`
+	NetworkSettings *NetworkSettings `json:"NetworkSettings"` // not implemented
+}
+
+// LogConfig holds the log information for a container
+type LogConfig struct {
+	Type   string            `json:"Type"`   // not implemented
+	Config map[string]string `json:"Config"` //idk type, not implemented
+}
+
+// RestarPolicy holds the container's policy on restart
+type RestartPolicy struct {
+	Name              string `json:"Name"`              // not implemented
+	MaximumRetryCount int    `json:"MaximumRetryCount"` //check type, not implemented
+}
+
+// CtrState represents the state of a container.
+type CtrState struct {
+	OciVersion string    `json:"OciVersion"`
+	Status     string    `json:"Status"`
+	Running    bool      `json:"Running"`
+	Paused     bool      `json:"Paused"`
+	Restarting bool      `json:"Restarting"` // not implemented
+	OOMKilled  bool      `json:"OOMKilled"`
+	Dead       bool      `json:"Dead"`
+	Pid        int       `json:"Pid"`
+	ExitCode   int32     `json:"ExitCode"`
+	Error      string    `json:"Error"` // not implemented
+	StartedAt  time.Time `json:"StartedAt"`
+	FinishedAt time.Time `json:"FinishedAt"`
+}
+
+// HostConfig represents the host configuration for the container
+type HostConfig struct {
+	Binds                []string                    `json:"Binds"`
+	ContainerIDFile      string                      `json:"ContainerIDFile"` //not implemented
```

Comment:
I think we do create this in kpod run - we can store it as an annotation and unmarshal it into here?